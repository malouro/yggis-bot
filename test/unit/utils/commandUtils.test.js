import {
	getArgumentsFromMessage,
	// getCommandFromMessage,
	// makeCommandFromModule,
} from '../../../src/utils/commands'

describe('Command Utilities', () => {
	describe('`getArgumentsFromMessage`', () => {
		test('gets arguments from a message', () => {
			const expectedLength = 6
			const result = getArgumentsFromMessage({
				content: `This message should have ${6} arguments.`,
			})

			expect(result).toBeInstanceOf(Array)
			expect(result).toHaveLength(expectedLength)
			expect(result).toMatchSnapshot()
		})

		test('returns empty array when message is empty or undefined', () => {
			const resultForEmptyMessage = getArgumentsFromMessage({ content: '' })
			const resultForNoMessage = getArgumentsFromMessage(undefined)

			expect(resultForEmptyMessage).toBeInstanceOf(Array)
			expect(resultForEmptyMessage).toHaveLength(0)

			expect(resultForNoMessage).toBeInstanceOf(Array)
			expect(resultForNoMessage).toHaveLength(0)
		})
	})
})
