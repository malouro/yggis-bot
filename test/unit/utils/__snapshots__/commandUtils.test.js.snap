// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command Utilities \`getArgumentsFromMessage\` gets arguments from a message 1`] = `
Array [
  "This",
  "message",
  "should",
  "have",
  "6",
  "arguments.",
]
`;
